model/map/src/main/java/org/keycloak/models/map/storage/tree/TreeStorageNodeInstance.java

```diff
@@ -0,0 +1,128 @@
+/*
```

Comment:
Several methods here are for the future use of tree store and do not have immediate use here. The number of such functions have been kept to the reasonable minimum.

Reply:
Perhaps the best to start the review from is from the tests, they have some extra descriptions of the use cases